src/MediatR/Mediator.cs

```diff
@@ -226,6 +256,18 @@ private IEnumerable<TWrapper> GetNotificationHandlers<TWrapper>(object notificat
                 .ToList();
         }
 
+        private Task<object> CreatePipeline(object request, RequestHandlerDelegate invokeHandler)
+        {
+            var behaviors = _multiInstanceFactory(typeof(IPipelineBehavior))

```

Comment:
This is really simple and clever. Almost _too_ simple? Have you considered a use case where one may want to select pipeline behaviors, such as based on the Type/Assembly of the request/handler?

But maybe it's as simple as putting that logic in the pipeline behavior's implementation as needed, since it has access to the request and response?